ngtools/local/handlers.py

```diff
@@ -14,6 +20,122 @@
 LOG = logging.getLogger(__name__)
 
 
+class AnnotationHandler(Handler):
+    """Handler that returns annotation data."""
+
+    annotation_cache = {}
+    spatical_cache = {}
+    grid_densities = [1, 2, 4, 8]
+
+    def read_from_file(self, path: str) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """Read annotation data from a given file. cache it as well as its segments."""
+        if path in self.annotation_cache:
+            return self.annotation_cache[path]
+        segments, bbox, offsets, affine = tap.load_from_file(path)
+        segments, offsets = tap.split_along_grid(
+            segments, bbox, [self.grid_densities[-1]]*3, offsets)
+        self.annotation_cache[path] = (segments, bbox, offsets, affine)
+        return segments, bbox, offsets, affine
+
+    def get(self, path: str) -> None:  # noqa: D102
+
+        if path.endswith("info"):
+            try:
+                info = self.get_info(path)
+            except Exception as e:
+                return self.send_error(404, e)
+            self.status = 200
+            self.headers["Content-type"] = "application/json"
+            self.body = json.dumps(info).encode()
+            return None
+        if self.parse_path(path) is not None:
+            try:
+                buffer = self.get_spaticals(path)
+            except Exception as e:
+                return self.send_error(404, e)
+            self.status = 200
+            self.headers["Content-type"] = "application/octet-stream"
+            self.body = buffer
+            return None
+        if path.endswith("transform.lta"):
+            try:
+                transform = self.get_transform(path)
+            except Exception as e:
+                return self.send_error(404, e)
+            self.status = 200
+            self.headers["Content-type"] = "text/plain"
+            self.body = transform.encode("utf-8")
+            return None
+        try:
+            folder_listing = {"path": path,
+                              "children": ["info", "0", "1", "2"]}
+            body = json.dumps(folder_listing).encode()
+            self.status = 200
+            self.headers.add_header("Content-Type", "application/json")
+            self.headers.add_header("Content-Length", str(len(body)))
+            self.body = body
+
+        except Exception as e:
+            return self.send_error(500, e)
+
+    def parse_path(self, path_str: str) -> Tuple[str, str, str, str] | None:
```

Comment:
Maybe rename `_parse_spatial_path` to make it clear the kind of path it expects.